### ISSUE TITLE
Rich push notifications not working with FCM.

### ISSUE THREAD

**Issue by netbe** [customerio/customerio-ios#159]
I am not able to make the rich push notifications work with the documentation details provided here: https://customer.io/docs/sdk/ios/rich-push/

First the current stable version `1.0.3`, is lacking methods when using FCM for rich methods. I've seen this is [fixed](https://github.com/customerio/customerio-ios/commit/0ca06183f9c0094019e3e58d0963b5931909c348) in latest beta.

But still the following code is not showing me the image in the notification (only the standard title and body are shown):

```
Messaging.shared.didReceive(request, withContentHandler: contentHandler)
```

Note:  the service is well started as I am able to modify the title or body manually

*Notification payload:*
```
{
    "message": {
        "apns": {
            "payload": {
                "CIO": {
                    "push": {
                        "link": "remote-habits://deep?message=hello&message2=world",
                        "image": "https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fpawsindoorssouthend.com%2Fwp-content%2Fuploads%2F2020%2F04%2Ftips-for-choosing-doggy-day-care.jpg"
                    }
                },
                "aps": {
                    "mutable-content": 1,
                    "alert": {
                        "title": "Title of your push goes here!",
                        "body": "Body of your push goes here!"
                    }
                }
            }
        }
    }
}
```

I am wondering if I need to initialise the `CustomerIO` instance within the service as well as it is independent from the app?

Any help appreciated.

**Comment by netbe** [comment]
Ok I made it work with initialising the customerIO instance inside the notification service + changing the image link (apparently the parsing of url failed?).

Do you have a date for the release of 1.1.0?

Note: I would advise to update your documentation on rich notifications of FCM to state it is not working on 1.0.3.

**Comment by levibostian** [comment]
Hello there, @netbe. Thanks for asking a question. I understand that it has been 5 days since you asked this question and have not received a response. You can always ask questions here on GitHub, but know that the fastest way to get a response from our team is through our customer support at `win@customer.io`. 

Now, to your question. 

> Ok I made it work with initialising the customerIO instance inside the notification service + changing the image link (apparently the parsing of url failed?).

Glad to hear that you found a solution. In your app's code, where are you initializing the CustomerIO instance? [I expected `application(didFinishLaunchingWithOptions:)` to be called in your `AppDelegate`](https://github.com/customerio/RemoteHabits-iOS/blob/419bb9949429af8fed1044c4afd1f987a378df0d/Remote%20Habits/AppDelegate.swift#L10-L21=) before the NotificationService got called but I guess it did not? 

Also, what did you change with your image link? We know that some image URL formats are not supported by the SDK today. We are looking at ways to improve this. If you provide to us some example URLs that do and do not work for notifications images that would be very helpful. You can send the URLs to `win@customer.io` if they are private. 

> Do you have a date for the release of 1.1.0?

[Our latest 1.1.0 beta](https://github.com/customerio/customerio-ios/releases/tag/1.1.0-beta.1) is considered stable enough for you to install into your app. We push beta builds to production once we see some customers are using it and not reporting issues to us. Feel free to install the latest beta to help us get the build into production. We do not have a date when 1.1 will go into production. The more people we see using it, the faster we feel confident in pushing it for you.

**Comment by netbe** [comment]
@levibostian thanks for the answer.

> Glad to hear that you found a solution. In your app's code, where are you initializing the CustomerIO instance? [I expected application(didFinishLaunchingWithOptions:) to be called in your AppDelegate](https://github.com/customerio/RemoteHabits-iOS/blob/419bb9949429af8fed1044c4afd1f987a378df0d/Remote%20Habits/AppDelegate.swift#L10-L21=) before the NotificationService got called but I guess it did not?

NotificationService has it's own bundle so it's not sharing code with the main app if I am not wrong. The CustomerIO instance should also be initialized or the site's info should be shared within a app-group.

About the URLs it's the one from the doc (using the duckduck.go search)

I am seeing you just released 5 days ago the 1.1.0 I'll give it a go again.

**Comment by levibostian** [comment]
I appreciate your reply back, @netbe. 

I documented this issue internally with our team so myself or someone else can look into this. I hope that we can either improve the SDK so that you don't need to initialize the SDK in the NotificationService or we update our docs help you not encounter this scenario. 

Again, thank you so much for reporting this issue. Greatly appreciated!
